web/src/client/storage.js

```diff
@@ -107,24 +105,47 @@ class StorageBaseClient {
   async calculateProposal({ candidateDevices, encryptionPassword, lvm, volumes }) {
     const proxy = await this.client.proxy(STORAGE_PROPOSAL_IFACE);
 
-    const dbusVolume = volume => {
-      return {
-        MountPoint: cockpit.variant("s", volume.mountPoint),
-        Encrypted: cockpit.variant("b", volume.encrypted),
-        FsType: cockpit.variant("s", volume.fsType),
-        MinSize: cockpit.variant("x", volume.minSize),
-        MaxSize: cockpit.variant("x", volume.maxSize),
-        FixedSizeLimits: cockpit.variant("b", volume.fixedSizeLimits),
-        Snapshots: cockpit.variant("b", volume.snapshots)
-      };
+    // Builds a new object without undefined attributes
+    const cleanObject = (object) => {
```

Comment:
Question: is this method useful in other clients? If so, I'd move to a better location. Let's discuss it later.

Reply:
I don't know. I would move it later if we find it useful in other places.